src/coreComponents/mesh/generators/WellGeneratorBase.cpp

```diff
@@ -521,62 +525,58 @@ void WellGeneratorBase::mergePerforations( array1d< array1d< localIndex > > cons
   }
 }
 
-void WellGeneratorBase::debugWellGeometry() const
+void WellGeneratorBase::logInternalWell() const
 {
-  if( MpiWrapper::commRank( MPI_COMM_GEOSX ) != 0 )
-  {
-    return;
-  }
-
-  std::cout << std::endl;
-  std::cout << "++++++++++++++++++++++++++" << std::endl;
-  std::cout << "WellGeneratorBase = " << getName() << std::endl;
-  std::cout << "MPI rank = " << MpiWrapper::commRank( MPI_COMM_GEOSX ) << std::endl << std::endl;
-  std::cout << "Number of well elements = " << m_numElems << std::endl;
-
+  TableData tableWellData;
   for( globalIndex iwelem = 0; iwelem < m_numElems; ++iwelem )
   {
-    std::cout << "Well element #" << iwelem << std::endl;
-    std::cout << "Coordinates of the element center: " << m_elemCenterCoords[iwelem] << std::endl;
-    if( m_nextElemId[iwelem] < 0 )
-    {
-      std::cout << "No next well element" << std::endl;
-    }
-    else
-    {
-      std::cout << "Next well element # = " << m_nextElemId[iwelem] << std::endl;
-    }
-    if( m_prevElemId[iwelem][0] < 0 )
-    {
-      std::cout << "No previous well element" << std::endl;
-    }
-    else
+    std::optional< globalIndex > nextElement;
+    std::optional< globalIndex > prevElement;
+
+    if( m_nextElemId[iwelem] >= 0 )
     {
-      std::cout << "Previous well element #" << m_prevElemId[iwelem][0] << std::endl;
+      nextElement =  m_nextElemId[iwelem];
     }
-    for( globalIndex inode = 0; inode < m_numNodesPerElem; ++inode )
+
+    if( m_prevElemId[iwelem][0] >= 0 )
     {
-      if( inode == 0 )
-      {
-        std::cout << "First well node: #" << m_elemToNodesMap[iwelem][inode] << std::endl;
-      }
-      else
-      {
-        std::cout << "Second well node: #" << m_elemToNodesMap[iwelem][inode] << std::endl;
-      }
+      prevElement =  m_prevElemId[iwelem][0];
     }
+
+    tableWellData.addRow( iwelem,
+                          m_elemCenterCoords[iwelem][0],
+                          m_elemCenterCoords[iwelem][1],
+                          m_elemCenterCoords[iwelem][2],
+                          prevElement,
+                          nextElement );
   }
 
-  std::cout << std::endl << "Number of perforations = " << m_numPerforations << std::endl;
+  string wellTitle = GEOS_FMT( "Well '{}' Element Table", getName() );
+  TableLayout tableWellLayout = TableLayout( {
+      TableLayout::ColumnParam{"Element no.", TableLayout::Alignment::right},
+      TableLayout::ColumnParam{"CoordX", TableLayout::Alignment::center},
+      TableLayout::ColumnParam{"CoordY", TableLayout::Alignment::center},
+      TableLayout::ColumnParam{"CoordZ", TableLayout::Alignment::center},
+      TableLayout::ColumnParam{"Prev\nElement", TableLayout::Alignment::right},
+      TableLayout::ColumnParam{"Next\nElement", TableLayout::Alignment::right},
+    }, wellTitle );
+
+  TableTextFormatter tableFormatter( tableWellLayout );
+  GEOS_LOG_RANK_0( tableFormatter.toString( tableWellData ));
```

Comment:
```suggestion
  string const wellTitle = GEOS_FMT( "Well '{}' Element Table", getName() );
  TableLayout const tableWellLayout = TableLayout( {
      TableLayout::ColumnParam{"Element no.", TableLayout::Alignment::right},
      TableLayout::ColumnParam{"CoordX", TableLayout::Alignment::center},
      TableLayout::ColumnParam{"CoordY", TableLayout::Alignment::center},
      TableLayout::ColumnParam{"CoordZ", TableLayout::Alignment::center},
      TableLayout::ColumnParam{"Prev\nElement", TableLayout::Alignment::right},
      TableLayout::ColumnParam{"Next\nElement", TableLayout::Alignment::right},
    }, wellTitle );

  TableTextFormatter const tableFormatter( tableWellLayout );
  GEOS_LOG_RANK_0( tableFormatter.toString( tableWellData ));
```
Does this compile? Should it compile?

Reply:
This is compiling, I  don't understand why it shouldn't compile

Reply:
When adding `const`, it's rarely obvious to say if it's going to compile without testing.